crates/bevy_text/src/pipeline.rs

```diff
@@ -51,30 +52,38 @@ impl<ID: Hash + Eq> TextPipeline<ID> {
     pub fn queue_text(
         &mut self,
         id: ID,
-        font_handle: Handle<Font>,
         fonts: &Assets<Font>,
-        text: &str,
-        font_size: f32,
+        sections: &[TextSection],
+        scale_factor: f64,
         text_alignment: TextAlignment,
         bounds: Size,
         font_atlas_set_storage: &mut Assets<FontAtlasSet>,
         texture_atlases: &mut Assets<TextureAtlas>,
         textures: &mut Assets<Texture>,
     ) -> Result<(), TextError> {
-        let font = fonts.get(font_handle.id).ok_or(TextError::NoSuchFont)?;
-        let font_id = self.get_or_insert_font_id(font_handle, font);
-
-        let section = SectionText {
-            font_id,
-            scale: PxScale::from(font_size),
-            text,
-        };
-
-        let scaled_font = ab_glyph::Font::as_scaled(&font.font, font_size);
+        let mut scaled_fonts = Vec::new();
+        let sections = sections
+            .iter()
+            .map(|section| {
+                let font = fonts.get(section.font.id).ok_or(TextError::NoSuchFont)?;
+                let font_id = self.get_or_insert_font_id(&section.font, font);
+                let font_size = scale_value(section.style.font_size, scale_factor);
+
+                scaled_fonts.push(ab_glyph::Font::as_scaled(&font.font, font_size));
+
+                let section = SectionText {
+                    font_id,
+                    scale: PxScale::from(section.style.font_size),
```

Comment:
this should be `PxScale::from(font_size)` to use the scaled value

Reply:
ah right, of course. [Fixed](https://github.com/bevyengine/bevy/pull/1245/commits/2d17c3d3705c91075112cb9a521c2eec8e1a72fc)